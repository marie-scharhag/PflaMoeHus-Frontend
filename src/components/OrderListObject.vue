<template>
  <div id="line">
    <div class="productobject">
      <div class="picture">
        <img v-bind:src="'/api/product/picture/'+ppath.id" alt="Picture" id="pic" />
      </div>
      <div class="information">
        <ul>
          <li id="prName">{{ pname }}</li>
          <li id="prNr">
            <span>Pnr: {{ particlenr }} </span>
          </li>
          <li id="prPrice">
            <span>{{ product[1] }} x {{ pprice }}€</span>
          </li>
          <li id="inTotal">
            <span
              >{{
                Math.round(pprice * product[1] * Math.pow(10, 2)) /
                Math.pow(10, 2)
              }}
              €</span
            >
          </li>
        </ul>
      </div>
    </div>
  </div>
</template>
<script lang="ts">
import "@/service/Product";
import { defineComponent, ref, computed } from "vue";
import { useProduct } from "../service/ProductStore";

export default defineComponent({
  name: "CartListObject",
  props: {
    product: Object,
  },

  setup(props) {
    const { getProductByArtNr } = useProduct();

    /**
     * finds the first picture of a product
     */
    const ppath = computed(() => {
      if (props.product)
        return (getProductByArtNr(props.product[0]) as Product).allPictures[0];
    });
    /**
     * name of the product
     */
    const pname = ref("");
    /**
     * price of the product
     */
    const pprice = ref(0);
    /**
     * articlenumber of the product
     */
    const particlenr = ref(0);
    /**
     * product
     */
    let p: Product = {
      articlenr: 0,
      version: 0,
      name: "",
      productType: "",
      roomType: "",
      price: 0,
      allPictures: [],
      height: 0,
      width: 0,
      depth: 0,
      allTags: [],
      available: 0,
      description: "",
      information: "",
    };

    /**
     * finds product and sets values
     */
    if (props.product) {
      p = getProductByArtNr(props.product[0]) as Product;
      pname.value = p.name;
      pprice.value = p.price;
      particlenr.value = p.articlenr;
    }

    return {
      ppath,
      pname,
      pprice,
      particlenr,
    };
  },
});
</script>
<style lang = "scss">
ul {
  list-style: none;
}
.information li {
  padding-top: 2.5%;
  padding-bottom: 2.5%;
  text-align: left;
}
#prName {
  font-size: 16px;
  font-weight: bold;
  padding-bottom: 0%;
}
#prPrice {
  font-size: 16px;
  padding: 0%;
}
#prNr {
  font-size: 12px;
  padding-bottom: 0%;
}
#inTotal {
  font-size: 16px;
  padding-top: 0%;
  padding-right: 0%;
  margin: 0;
  float: right;
}
#amount {
  float: left;
  font-size: 14px;
}
.productobject {
  margin-left: 2em;
  margin-right: 2em;
  margin-top: 1.5em;
  margin-bottom: 1.5em;
  min-width: 35em;
  width: 100%;
  float: left;
}
#delete {
  border-style: none;
  background: none;
  &:focus {
    outline: none;
  }
}

#pic {
  float: left;
  width: 5em;
}
#line {
  border-bottom: 2px solid $color-grey2;
}

.information {
  float: left;
  margin-left: 0;
  width: 80%;
}

.close {
  float: right;
}
</style>