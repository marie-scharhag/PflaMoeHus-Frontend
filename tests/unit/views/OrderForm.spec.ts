import { mount, shallowMount } from "@vue/test-utils"

import OrderForm from "@/views/OrderForm.vue"
import OrderListObject from "@/views/OrderForm.vue"

describe('Orderform.vue test',() => {

    const cmp = shallowMount(OrderForm)

    test('HTML + structur snapshot test',() => {
        expect(cmp.html()).toMatchSnapshot()
        expect(cmp).toMatchSnapshot()
    })

    test('if adresses, credit-/bankcards  exist',() => {
        expect(cmp.html()).toContain('adress')
        expect(cmp.html()).toContain('bankcard')
        expect(cmp.html()).toContain('creditcard')
    })

    test('if component orderlistobject exist + is used',() => {
        expect(cmp.findComponent(OrderListObject)).toBeTruthy()
        expect(() => cmp.getComponent('.not-there')).toThrowError()
    })

    test('Test if error works (there should not be one)',() => {
        const wrapper = mount(OrderForm)
        expect(wrapper.html()).not.toContain(`streetnameerror`)
    })
    
})