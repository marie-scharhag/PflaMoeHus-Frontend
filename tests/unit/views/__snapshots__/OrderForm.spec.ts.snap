// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orderform.vue test HTML + structur snapshot test 1`] = `
<div class="orderForm">
  <h1>Bestellübersicht</h1>
  <form>
    <div class="adress">
      <h2>Versandadresse</h2>
      <!--v-if-->
      <div class="row"><label for="street" class="col7">Straße</label></div>
      <div class="row">
        <div class="error"></div><input id="street" type="text" class="col6" required="">
      </div>
      <div class="row"><label for="num" class="col1">Hausnummer</label></div>
      <div class="row">
        <div class="error"></div><input id="num" type="text" class="col1" required="">
      </div>
      <div class="row"><label for="num" class="col1">Postleitzahl</label></div>
      <div class="row">
        <div class="error"></div><input id="num" type="text" class="col1" required="">
      </div>
      <div class="row"><label for="street" class="col7">Stadt</label></div>
      <div class="row">
        <div class="error"></div><input id="street" type="text" class="col6" required="">
      </div>
    </div>
    <h2>Zahlungsart</h2>
    <div class="payment">
      <div class="row">
        <p class="error"></p><input id="creditcard" type="radio" value="creditcard"><label for="creditcard">Kredit- oder Debitkarte</label>
      </div>
      <div class="creditC">
        <div class="row"><label for="cardnumber" class="col3">Kartennummer</label><input type="number"></div>
        <div class="error"></div>
        <div class="row"><label for="cardholder" class="col3">Name des Karteninhabers</label><input type="text"></div>
        <div class="error"></div>
        <div class="row"><label for="expires" class="col3">Ablaufdatum</label><select>
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
            <option value="6">6</option>
            <option value="7">7</option>
            <option value="8">8</option>
            <option value="9">9</option>
            <option value="10">10</option>
            <option value="11">11</option>
            <option value="12">12</option>
          </select><select>
            <option value="2021">2021</option>
            <option value="2022">2022</option>
            <option value="2023">2023</option>
            <option value="2024">2024</option>
            <option value="2025">2025</option>
            <option value="2026">2026</option>
            <option value="2027">2027</option>
            <option value="2028">2028</option>
            <option value="2029">2029</option>
            <option value="2030">2030</option>
            <option value="2031">2031</option>
          </select>
          <div class="error"></div>
        </div>
      </div>
      <div class="row"><input id="bankcard" type="radio" value="bankcard"><label for="bankcard">Bankkarte</label></div>
      <div class="bankC">
        <div class="row"><label for="iban" class="col3">IBAN</label><input id="iban" type="text"></div>
        <div class="error"></div>
        <div class="row"><label for="bank" class="col3">Bank</label><input id="bank" type="text"></div>
        <div class="error"></div>
        <div class="row"><label for="cardholder" class="col3">Name des Karteninhabers</label><input type="text"></div>
        <div class="error"></div>
      </div>
    </div>
  </form>
  <h2>Vorausichtliches Lieferdatum: 19.2.2021</h2>
  <div class="error"></div>
  <div class="error"></div>
  <div class="row"></div>
  <div class="row">
    <hr>
  </div>
  <div class="row">
    <p class="price">0€</p>
  </div><br>
  <div class="row">
    <div id="other" class="error"></div><input type="submit" value="Bestellen">
  </div>
</div>
`;

exports[`Orderform.vue test HTML + structur snapshot test 2`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="orderForm"
      >
        <h1>
          Bestellübersicht
        </h1>
        <form>
          <div
            class="adress"
          >
            <h2>
              Versandadresse
            </h2>
            <!--v-if-->
            <div
              class="row"
            >
              <label
                class="col7"
                for="street"
              >
                Straße
              </label>
            </div>
            <div
              class="row"
            >
              <div
                class="error"
              />
              <input
                class="col6"
                id="street"
                required=""
                type="text"
              />
            </div>
            <div
              class="row"
            >
              <label
                class="col1"
                for="num"
              >
                Hausnummer
              </label>
            </div>
            <div
              class="row"
            >
              <div
                class="error"
              />
              <input
                class="col1"
                id="num"
                required=""
                type="text"
              />
            </div>
            <div
              class="row"
            >
              <label
                class="col1"
                for="num"
              >
                Postleitzahl
              </label>
            </div>
            <div
              class="row"
            >
              <div
                class="error"
              />
              <input
                class="col1"
                id="num"
                required=""
                type="text"
              />
            </div>
            <div
              class="row"
            >
              <label
                class="col7"
                for="street"
              >
                Stadt
              </label>
            </div>
            <div
              class="row"
            >
              <div
                class="error"
              />
              <input
                class="col6"
                id="street"
                required=""
                type="text"
              />
            </div>
          </div>
          <h2>
            Zahlungsart
          </h2>
          <div
            class="payment"
          >
            <div
              class="row"
            >
              <p
                class="error"
              />
              <input
                id="creditcard"
                type="radio"
                value="creditcard"
              />
              <label
                for="creditcard"
              >
                Kredit- oder Debitkarte
              </label>
            </div>
            <div
              class="creditC"
            >
              <div
                class="row"
              >
                <label
                  class="col3"
                  for="cardnumber"
                >
                  Kartennummer
                </label>
                <input
                  type="number"
                />
              </div>
              <div
                class="error"
              />
              <div
                class="row"
              >
                <label
                  class="col3"
                  for="cardholder"
                >
                  Name des Karteninhabers
                </label>
                <input
                  type="text"
                />
              </div>
              <div
                class="error"
              />
              <div
                class="row"
              >
                <label
                  class="col3"
                  for="expires"
                >
                  Ablaufdatum
                </label>
                <select>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9
                  </option>
                  <option
                    value="10"
                  >
                    10
                  </option>
                  <option
                    value="11"
                  >
                    11
                  </option>
                  <option
                    value="12"
                  >
                    12
                  </option>
                </select>
                <select>
                  <option
                    value="2021"
                  >
                    2021
                  </option>
                  <option
                    value="2022"
                  >
                    2022
                  </option>
                  <option
                    value="2023"
                  >
                    2023
                  </option>
                  <option
                    value="2024"
                  >
                    2024
                  </option>
                  <option
                    value="2025"
                  >
                    2025
                  </option>
                  <option
                    value="2026"
                  >
                    2026
                  </option>
                  <option
                    value="2027"
                  >
                    2027
                  </option>
                  <option
                    value="2028"
                  >
                    2028
                  </option>
                  <option
                    value="2029"
                  >
                    2029
                  </option>
                  <option
                    value="2030"
                  >
                    2030
                  </option>
                  <option
                    value="2031"
                  >
                    2031
                  </option>
                </select>
                <div
                  class="error"
                />
              </div>
            </div>
            <div
              class="row"
            >
              <input
                id="bankcard"
                type="radio"
                value="bankcard"
              />
              <label
                for="bankcard"
              >
                Bankkarte
              </label>
            </div>
            <div
              class="bankC"
            >
              <div
                class="row"
              >
                <label
                  class="col3"
                  for="iban"
                >
                  IBAN
                </label>
                <input
                  id="iban"
                  type="text"
                />
              </div>
              <div
                class="error"
              />
              <div
                class="row"
              >
                <label
                  class="col3"
                  for="bank"
                >
                  Bank
                </label>
                <input
                  id="bank"
                  type="text"
                />
              </div>
              <div
                class="error"
              />
              <div
                class="row"
              >
                <label
                  class="col3"
                  for="cardholder"
                >
                  Name des Karteninhabers
                </label>
                <input
                  type="text"
                />
              </div>
              <div
                class="error"
              />
            </div>
          </div>
        </form>
        <h2>
          Vorausichtliches Lieferdatum: 19.2.2021
        </h2>
        <div
          class="error"
        />
        <div
          class="error"
        />
        <div
          class="row"
        >
          
          
        </div>
        <div
          class="row"
        >
          <hr />
        </div>
        <div
          class="row"
        >
          <p
            class="price"
          >
            0€
          </p>
        </div>
        <br />
        <div
          class="row"
        >
          <div
            class="error"
            id="other"
          />
          <input
            type="submit"
            value="Bestellen"
          />
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": Object {},
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": Object {},
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.5",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "Field": Object {
      "bank": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "city": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "cowner": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "creditcardnumber": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "dateOfExpiry": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "houseNumber": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "iban": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "owner": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "postCode": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "streetName": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
    },
    "FieldErrors": Object {
      "bank": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "city": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "cowner": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "creditcardnumber": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "dateOfExpiry": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "houseNumber": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "iban": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "other": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "owner": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "postCode": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
      "streetName": RefImpl {
        "__v_isRef": true,
        "_rawValue": "",
        "_shallow": false,
        "_value": "",
      },
    },
    "__emitted": Object {},
    "adrChange": [Function],
    "adresses": Array [],
    "dateOfExpiry": undefined,
    "dateOfExpiryMonth": 1,
    "dateOfExpiryYear": 2021,
    "deliveryDate": "19.2.2021",
    "inTotal": 0,
    "isHidden": true,
    "notavailableerror": "",
    "notavailableerrorempty": "",
    "orderwassent": false,
    "payment": "",
    "paymenterror": "",
    "productList": Array [],
    "selectedadr": "",
    "sendOrder": [Function],
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
